src/components/LoginForm/LoginForm.jsx

```diff
@@ -1,95 +1,114 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
-
+import { useForm } from 'react-hook-form';
 import { handleRegisterAndLogin } from '@/utils/loginHandlers';
 
 import styles from './LoginForm.module.css';
 
 export default function LoginForm() {
-    /* eslint-disable-next-line */
+    const navigate = useNavigate();
+
+    /* State to manage the response data from an API request */
     const [response, setResponse] = useState({
         loading: false,
         data: {},
         error: null,
     });
-    const [loginOrRegisterData, setLoginOrRegisterData] = useState({ email: '', password: '' });
 
-    const navigate = useNavigate();
+    /* Triggers a re-rendering for showing the invalid email or password feedback */
+    useEffect(() => {
+        if (response.error) {
+            setError('password', { type: 'manual', message: 'Invalid email or password' });
+        }
+    }, [response]);
+
+    /* Destructure variables from the useForm hook for form management */
+    const {
+        register,
+        handleSubmit,
+        formState: { errors },
+        setError,
+    } = useForm();
 
     // const email = 'wewe@wewe.com';
     // const password = 'Hola123$';
 
-    const handleSubmit = async (e) => {
-        e.preventDefault();
```

Comment:
why e.preventDefault() is removed?

Reply:
Because I'm using the handleSubmit function from react hook form